### PR TITLE
[MBL-16831][Student] Offline all courses

### PR DESCRIPTION
**Pull request by tamaskozmer**
Test plan: Design and expected behavior is in the Figma link attached to the ticket.
- There were some changes which affected both apps (changing some titles, and disabling the stars in offline mode).
- Smoke test the feature in the Teacher app as well.

refs: MBL-16831
affects: Student
release note: none

## Checklist

- [x] Run E2E test suite
- [x] Tested in dark mode
- [x] Tested in light mode

### CODE REVIEW

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by tamaskozmer** [comment]
Successful Student ui-test run: https://app.bitrise.io/build/e22cc003-086b-4169-96a2-3ba45f04cd56

**Comment by kdeakinstructure** [comment]
> Successful Student ui-test run: https://app.bitrise.io/build/e22cc003-086b-4169-96a2-3ba45f04cd56

Can you link a successful E2E run as well please? I saw you have ticked the checkbox, but I cannot see the results.

**Review by kdeakinstructure — APPROVED** [review]
QA 👍

**Comment by tamaskozmer** [comment]
> > Successful Student ui-test run: https://app.bitrise.io/build/e22cc003-086b-4169-96a2-3ba45f04cd56
> 
> Can you link a successful E2E run as well please? I saw you have ticked the checkbox, but I cannot see the results.

You are right, I haven't linked it, because it was not successful. 😄 However I fixed all the tests that were broken by these changes. I ran one on the offline branch as well and 1 test is failing there as well, so we will fix that later.

File: apps/student/src/main/java/com/instructure/student/features/dashboard/edit/datasource/StudentEditDashboardLocalDataSource.kt [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import com.instructure.pandautils.room.offline.entities.EnrollmentState
+import com.instructure.pandautils.room.offline.facade.CourseFacade
+
+class StudentEditDashboardLocalDataSource(
+    private val courseFacade: CourseFacade,
+    private val editDashboardItemDao: EditDashboardItemDao
+) : StudentEditDashboardDataSource {
+
+    override suspend fun getCourses(): List<List<Course>> {
+        val courseMapper: suspend (EditDashboardItemEntity) -> Course = {
```
Comment (hermannakos) [inline apps/student/src/main/java/com/instructure/student/features/dashboard/edit/datasource/StudentEditDashboardLocalDataSource.kt:32]: hmm, nice

File: libs/pandautils/src/main/java/com/instructure/pandautils/room/offline/daos/EditDashboardItemDao.kt [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import androidx.room.Query
+import androidx.room.Transaction
+import androidx.room.Update
+import com.instructure.pandautils.room.offline.entities.EditDashboardItemEntity
+import com.instructure.pandautils.room.offline.entities.EnrollmentState
+
+@Dao
+interface EditDashboardItemDao {
+
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
```
Comment (hermannakos) [inline libs/pandautils/src/main/java/com/instructure/pandautils/room/offline/daos/EditDashboardItemDao.kt:32]: I would add a test for the OnConflictStrategy just to make sure we don't modify it by accident.

Comment (tamaskozmer) [reply libs/pandautils/src/main/java/com/instructure/pandautils/room/offline/daos/EditDashboardItemDao.kt:32]: We always use it only after dropping all records, but I can do that.

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Pull request merged**